pkg/chains/signing.go

```diff
@@ -58,21 +62,45 @@ type TaskRunSigner struct {
 }
 
 // IsSigned determines whether a TaskRun has already been signed.
-func IsSigned(tr *v1beta1.TaskRun) bool {
+func IsSigned(tr *v1beta1.TaskRun) (bool, string) {
 	signature, ok := tr.ObjectMeta.Annotations[ChainsAnnotation]
 	if !ok {
-		return false
+		return false, ""
 	}
-	return signature == "true"
+	return signature == "true", signature
+}
+
+// IsRetrying determines whether a TaskRun is retrying state.
+func IsRetrying(tr *v1beta1.TaskRun) (bool, string) {
+	retries, ok := tr.ObjectMeta.Annotations[ChainsRetriesAnnotation]
+	if !ok {
+		return false, ""
+	}
+	return true, retries
 }
 
 // MarkSigned marks a TaskRun as signed.
-func MarkSigned(tr *v1beta1.TaskRun, ps versioned.Interface, annotations map[string]string) error {
+func MarkSigned(tr *v1beta1.TaskRun, ps versioned.Interface, annotations map[string]string,
+	state string) error {
+	return markannotation(tr, ps, annotations, ChainsAnnotation, state)
+}
+
+// MarkRetries marks a TaskRun as retrying.
+func MarkRetries(tr *v1beta1.TaskRun, ps versioned.Interface, annotations map[string]string,
```

Comment:
Instead of just marking retries, we could call this `AddRetry`, and handle increasing the count here along with updating the annotation

Reply:
Changed it to AddRetry.